### PR TITLE
Process negative indices for lrange

### PR DESCRIPTION
**Pull request by kappa**
With tests.

### CODE REVIEW

File: t/09-list.t [hunk 1]
```diff
@@ -16,7 +16,7 @@ x   LLEN
 x   LPOP
 x   LPUSH
 x   LPUSHX
-    LRANGE
+o   LRANGE
```
Comment (jlavallee) [inline t/09-list.t]: What about LRANGE remains to be tested or implemented?

File: lib/Test/Mock/Redis.pm [hunk 2]
```diff
@@ -463,7 +463,14 @@ sub llen {
 sub lrange {
     my ( $self, $key, $start, $end ) = @_;
 
-    return @{ $self->_stash->{$key} }[$start..$end];
+    my $list = $self->_stash->{$key};
+
+    $start < 0 and $start = @$list + $start;
```
Comment (jlavallee) [inline lib/Test/Mock/Redis.pm]: I can't say I'm a huge fan of using "and" for flow control like this.  Would you mind writing it using if blocks?

**Comment by jlavallee** [comment]
Thanks for implementing this!  I've left a couple of minor comments inline.

**Comment by jplindstrom** [comment]
I like this feature. Is there anything I can do to help get this merged? (like rewriting the conditionals, if that's the only sticking point)

**Comment by jlavallee** [comment]
I left two comments, neither of which has been responded to.

If you'd like to make an alternate pull request, I'd be happy to take a look.  Thanks!

**Comment by jplindstrom** [comment]
This is superseded by #23, so this could be closed if that PR could be merged and released.

**Pull request closed without merging**